src/main/java/seedu/main/Parser.java

```diff
@@ -0,0 +1,165 @@
+package seedu.main;
+
+import seedu.command.Command;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * The class {@code Parser} is responsible for parsing user commands into {@code Command} objects.
+ * It manages the registration of commands and the extraction of arguments for each command.
+ *
+ * @@author DanLinhHuynh-Niwashi-reuse
+ * with minimal changes
+ */
+public class Parser {
+    /** Map that associates command words with their corresponding {@code Command} objects */
+    private static Logger logger = Logger.getLogger("Parser");
+    private Map<String, Command> commands = new LinkedHashMap<>();
+
+    /**
+     * Getter for the map of registered commands.
+     *
+     * @return A map of command words to {@code Command} objects.
+     */
+    public Map<String, Command> getCommands() {
+        return commands;
+    }
+
+    /**
+     * Registers a {@code Command} with its corresponding command word.
+     *
+     * @param command The {@code Command} object to be registered.
+     */
+    public void registerCommands(Command command) {
+        String commandWord = null;
+        try {
+            commandWord = (String) command.getClass().getField("COMMAND_WORD").get(null);
+
+            assert commandWord != null : "Command word should not be null.";
+
+            logger.log(Level.INFO, "Adding..." + commandWord);
+            commands.put(commandWord, command);
+        } catch (IllegalAccessException | NoSuchFieldException e) {
+            // Ignored
+        }
+    }
+
+    /**
+     * Parses a command string into a {@code Command} and assign its arguments.
+     *
+     * @param commandString The command string entered by the user.
+     * @return The {@code Command} associated with the command string, with assigned arguments.
+     */
+    public Command parseCommand(String commandString) {
```

Comment:
This function seems to be handling both parsing and argument extraction You can consider how to update this to suit SRP

Reply:
Created a new issue. Thank you!